File: src/main/java/com/example/hexagonal/application/port/output/EventPublisher.java
package com.example.hexagonal.application.port.output;

import com.example.hexagonal.domain.event.DomainEvent;

import java.util.List;

/**
 * Output Port - 事件發布者介面。
 * <p>
 * 遵循 OCP：新增事件類型不需要修改此介面。
 * 遵循 DIP：Application 層定義介面，Adapter 層實作。
 */
public interface EventPublisher {
    void publish(DomainEvent event);
    void publishAll(List<DomainEvent> events);
}

File: src/application/ports/output/EventPublisher.ts
// ============================================================
// Output Port - 事件發布者介面
// ============================================================
// 遵循 DIP：應用層定義事件發布的抽象介面，
// 具體的訊息佇列（Kafka, RabbitMQ 等）實作在 Adapter 層。
//
// 遵循 OCP：新增事件類型時，不需要修改發布者介面。
// ============================================================

import { DomainEvent } from '../../domain/events/DomainEvent';

export interface EventPublisher {
  publish(event: DomainEvent): Promise<void>;
  publishAll(events: DomainEvent[]): Promise<void>;
}
